examples/sync/src/bin/client.rs

```diff
@@ -0,0 +1,253 @@
+//! This client demonstrates how to use the ADB sync functionality
+//! to synchronize to the server's state. It fetches server metadata
+//! to determine sync parameters and then performs the actual sync operation.
+
+use clap::{Arg, Command};
+use commonware_cryptography::sha256::Digest;
+use commonware_runtime::{tokio as tokio_runtime, Metrics as _, Runner};
+use commonware_storage::{
+    adb::any::sync::{self, client::Config as SyncConfig},
+    mmr::hasher::Standard,
+};
+use commonware_sync::{crate_version, create_adb_config, Database, NetworkResolver};
+use std::{
+    net::{Ipv4Addr, SocketAddr},
+    num::NonZeroU64,
+};
+use tracing::{error, info};
+
+/// Default server address.
+const DEFAULT_SERVER: &str = "127.0.0.1:8080";
+
+#[derive(Debug)]
+struct ClientConfig {
+    /// Server address to connect to.
+    server: SocketAddr,
+    /// Batch size for fetching operations.
+    batch_size: u64,
+    /// Storage directory.
+    storage_dir: String,
+    /// Port on which metrics are exposed.
+    metrics_port: u16,
+}
+
+#[derive(Debug)]
+struct ServerMetadata {
+    target_hash: Digest,
+    oldest_retained_loc: u64,
+    latest_op_loc: u64,
+}
+
+/// Get server metadata to determine sync parameters.
+async fn get_server_metadata<E>(
+    resolver: &NetworkResolver<E>,
+) -> Result<ServerMetadata, Box<dyn std::error::Error>>
+where
+    E: commonware_runtime::Network + Clone,
+{
+    info!("Requesting server metadata...");
+
+    let metadata = resolver.get_server_metadata().await?;
+    let metadata = ServerMetadata {
+        target_hash: metadata.target_hash,
+        oldest_retained_loc: metadata.oldest_retained_loc,
+        latest_op_loc: metadata.latest_op_loc,
+    };
+    info!(?metadata, "Received server metadata");
+    Ok(metadata)
+}
+
+/// Create a new database synced to the server's state.
+async fn sync<E>(
+    context: E,
+    resolver: NetworkResolver<E>,
+    config: &ClientConfig,
+) -> Result<Database<E>, Box<dyn std::error::Error>>
+where
+    E: commonware_runtime::Storage
+        + commonware_runtime::Clock
+        + commonware_runtime::Metrics
+        + commonware_runtime::Network,
+{
+    info!(server = %config.server, "Starting ADB sync from server");
```

Comment:
For the user, I think wed can just call this `Starting sync from server` (don't ned ADB here)